torchvision/csrc/custom_ops/custom_ops.cpp

```diff
@@ -6,9 +6,139 @@
 
 using namespace at;
 
+using torch::Tensor;
+using torch::autograd::AutogradContext;
+using torch::autograd::Variable;
+using torch::autograd::variable_list;
+
+class ROIAlignFunction : public torch::autograd::Function<ROIAlignFunction> {
+ public:
+  static variable_list forward(
+      AutogradContext* ctx,
+      Variable input,
+      Variable rois,
+      const double spatial_scale,
+      const int64_t pooled_height,
+      const int64_t pooled_width,
+      const int64_t sampling_ratio) {
+    ctx->saved_data["spatial_scale"] = spatial_scale;
+    ctx->saved_data["pooled_height"] = pooled_height;
+    ctx->saved_data["pooled_width"] = pooled_width;
+    ctx->saved_data["sampling_ratio"] = sampling_ratio;
+    ctx->saved_data["batch_size"] = input.size(0);
```

Comment:
nit: This is fine as is, this is more a question for myself: I thought `IValue` allowed us to to also have lists of ints internally, is that right?

Reply:
Right now it was a 1-1 transcription of the Python Function. I'll check.
If we are going to keep the backward signature as is (as opposed to changing it to IntList), the benefit will be marginal, but given that we're providing an example, too...

Reply:
Works well, I switched this to `input_shape`, but we might change the function signature when we remove the Python extension.